azure/services/async/async.go

```diff
@@ -230,3 +244,9 @@ func getRetryAfterFromError(err error) time.Duration {
 	}
 	return ret
 }
+
+// ContextDeadlineExceededOrCanceled checks if it's an intermittent context deadline
+// exceeded or canceled error.
+func ContextDeadlineExceededOrCanceled(ctx context.Context) bool {
```

Comment:
```suggestion
func IsContextDeadlineExceededOrCanceledError(ctx context.Context) bool {
```
is a bit of a mouthful but more consistent with

```
func IsOperationNotDoneError(target error) bool 
```

Reply:
if we're not checking the error itself, it should be

```suggestion
func IsContextDeadlineExceededOrCanceled(ctx context.Context) bool {
```